hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -391,12 +411,15 @@ def register_profile(name, parent=None, **kwargs):
         parent (or settings.default, if parent is None).
         """
         if not isinstance(name, (str, text_type)):
```

Comment:
It would be nice to replace the `(str, text_type)` with `compat.string_types`, here and in the two other cases.